main.js

```diff
@@ -1,237 +1,74 @@
+const searchBarTarget = document.getElementById('searchBarTarget');
+const searchTextTarget = document.getElementById('searchTextTarget');
 
 const api = new API();
 
-const init = () => {
-  
-  ////////////// Best
-  const getBest = async () => {
-    const data = await api.getItem({ type : 'best' });
-    if(data) renderBest(data);
-  };
-  const renderBest = ({prefix, list}) => bestItemTarget.innerHTML = `<a class="carousel__item"><img src="${prefix}${list[0].src}"></a>`;
-
-  const moveCarousel = (target, transition = true) => {
-    if(target.dataset.arrow === 'prev') {
-      if(transition) {
-        panel.classList.add('transition-on', 'move-prev');
-      }
-      setTimeout(() => {
-        const list = panel.querySelectorAll('.panel__item');
-        panel.insertBefore(list[list.length - 1], list[0]);
-        pagingHover.insertBefore(pagingHover.firstElementChild, null);
-        panel.classList.remove('transition-on', 'move-prev');
-      }, 300);
-    }
-    if(target.dataset.arrow === 'next') {
-      if(transition) {
-        panel.classList.add('transition-on', 'move-next');
-      }
-      setTimeout(() => {
-        const list = panel.querySelectorAll('.panel__item');
-        panel.insertBefore(list[0], null);
-        pagingHover.insertBefore(pagingHover.lastElementChild, pagingHover.firstElementChild);
-        panel.classList.remove('transition-on', 'move-next');
-      }, 300);
-    }
-  }
-
-  pagingHover.addEventListener('click', ({ target }) => {
-    if(!target.classList.contains('active')) {
-      const list = pagingHover.querySelectorAll('div');
-      let activeIndex = 0;
-      let targetIndex = 0;
-      for(let i = 0; i < list.length; i++) {
-        if(list[i].classList.contains('active')) {
-          activeIndex = i;
-        }
-        if(list[i] === target) {
-          targetIndex = i;
-        }
-      }
-      let sum = activeIndex - targetIndex;
-      let str = 'prev';
-      if(sum < 0) {
-        str = 'next';
-        sum *= -1;
-      }
-      while(sum-- > 0) {
-        if(str === 'prev') {
-          const list = panel.querySelectorAll('.panel__item');
-          panel.insertBefore(list[list.length - 1], list[0]);
-          pagingHover.insertBefore(pagingHover.firstElementChild, null);
-        } else {
-          const list = panel.querySelectorAll('.panel__item');
-          panel.insertBefore(list[0], null);
-          pagingHover.insertBefore(pagingHover.lastElementChild, pagingHover.firstElementChild);
-        }
-      }
-    }
+const init = async () => {
+  const keywords = await api.getKeyword();
+  createRollKeyword({ target : rollKeywordTarget, keywords });
+  createBest(bestItemTarget);
+  createEvent({ target : eventTarget, pagingTarget : eventPagingTarget });
+  createBox({ target : boxTarget, buttonTarget : moreButtonTarget });
+  createCarouselBox({ target : carouselBoxTarget, pagingTarget : carouselBoxPagingTarget });
+  createSearchList({ 
```

Comment:
createXXX 함수가 인스턴스 생성 후에 `init`메서드를 실행하고 있네요. `init`메서드가 외부에 공개 되어 사용되는 의미있는 경우가 아직은 없어 보이네요. 생성자 함수에서 멤버 변수 초기화 후에 `init`메서드를 실행하도록 하여 인스턴스 생성만 하면 되도록 변경하면 어떨까요?

```suggestion
  const [keywords, bestData, eventData, carouselData] = await Promise.all([
    api.getKeyword(),
    api.getItem({ type : 'best' }),
    api.getItem({ type : 'event' }),
    api.getItem({ type : 'carousel' })
  ]);
  new RollKeyword({target, keywords});
  new Best({ target, ...bestData });
  new Event({ target, pagingTarget, ...eventData });
  new Box({ target, buttonTarget });
  new CarouselBox({ target, pagingTarget, ...carouselData });
  new Search(args);
```

Reply:
> createXXX 함수가 인스턴스 생성 후에 `init`메서드를 실행하고 있네요. `init`메서드가 외부에 공개 되어 사용되는 의미있는 경우가 아직은 없어 보이네요. 생성자 함수에서 멤버 변수 초기화 후에 `init`메서드를 실행하도록 하여 인스턴스 생성만 하면 되도록 변경하면 어떨까요?

 main이 훨씬 깔끔해 보이네요!
제안해주신 코드가 너무 아름답습니다... 💓